src/cljs/main/broadfcui/common/method/config_io.cljs

```diff
@@ -51,20 +51,21 @@
        (filter #(string/blank? ((keyword (:name %)) outputs))
                (:outputs inputs-outputs))))
    :-add-default-outputs
-   (fn [{:keys [props state this]}]
+   (fn [{:keys [after-update props locals state this]}]
      (let [{:keys [begin-editing]} props
-           {:keys [outputs editing?]} @state
+           {:keys [editing?]} @state
            new-outputs (->> (this :-get-defaultable-outputs)
                             (map (fn [{:keys [name]}] [(keyword name) (str "this." (-> name (string/split ".") last))]))
-                            (into {})
-                            (merge outputs))]
-       (when-not editing? (begin-editing))
-       (swap! state update :outputs merge new-outputs)))
+                            (into {}))
+           new-outputs-vector (into [] new-outputs)]
+       (do (when-not editing? (begin-editing))
+           (after-update #(doseq [output new-outputs-vector] (((key output) @locals) :set-value (val output))))
```

Comment:
You could destructure `output` in the binding:
```clojure
(doseq [[k v] new-outputs-vector]
   ((k @locals) :set-value v))
```
And you should be able to `doseq` across a map in the same way, so you don't need to convert to a vector first.

Reply:
that looks so much better! :D